pkg/crd/gen.go

```diff
@@ -181,6 +185,87 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 	return nil
 }
 
+func removeDescriptionFromMetadata(crd *apiext.CustomResourceDefinition) {
+	for _, versionSpec := range crd.Spec.Versions {
+		if versionSpec.Schema != nil {
+			removeDescriptionFromMetadataProps(versionSpec.Schema.OpenAPIV3Schema)
+		}
+	}
+}
+
+func removeDescriptionFromMetadataProps(v *apiext.JSONSchemaProps) {
+	if v == nil {
+		return
+	}
+	for str, property := range v.Properties {
+		if str == "metadata" {
+			if property.Description != "" {
+				fmt.Fprintf(os.Stderr, "Warning: metadata description unsupported. Removing description: %s\n", property.Description)
+				property.Description = ""
+			}
+			if property.Items != nil {
+				removeDescriptionFromMetadataProps(property.Items.Schema)
+				for i := range property.Items.JSONSchemas {
+					removeDescriptionFromMetadataProps(&property.Items.JSONSchemas[i])
+				}
+			}
+			if property.AdditionalProperties != nil {
+				removeDescriptionFromMetadataProps(property.AdditionalProperties.Schema)
+			}
+			if property.AdditionalItems != nil {
+				removeDescriptionFromMetadataProps(property.AdditionalItems.Schema)
+			}
+			// property var reference is fine -- we handle the persistence of the modfications on the line below
+			//nolint:gosec
+			removeDescriptionFromMetadataProps(&property)
```

Comment:
why are we recursing here?

Reply:
See feedback from @whalecold [here](https://github.com/kubernetes-sigs/controller-tools/pull/511/commits/cea95b3c509d4c30168e91d59f0a1e4eada49390#r530237711)

I took it to mean that there could be objects embedded within the object (e.g. within the spec field), is that wrong? See the first commit for the non-recursive approach.

@whalecold can you provide an example of where this breaks if you don't recurse the fields that contain addition `JSONSchemaProps`?

Reply:
Reverted back to modifying only the top level metadata description, as @DirectXMan12 and I tested that kube apiserver accepts embedded objects with metadata descriptions.